docs/design/user_group_enforcement.md

```diff
@@ -0,0 +1,478 @@
+
+#### User
+
+Users that are specified as part of a limit will be assigned that specific limit. 
+If a limit with a wildcard is specified all users that are not already assigned a limit will be assigned the wildcard limit.
+For the user entry the wildcard sets a default quota for every user on that queue. Overridden by the named user entries.
+
+Specifying a single limit for users using the wildcard is allowed. There is no requirement to have a wildcard entry.
+
+#### Group
+
+For groups the interpretation is slightly more complex. 
+As explained as part of the tracking documentation the group that the usage is tracked against must match the membership of the user. 
+A wildcard group matches any group.
+
+Specifying a wildcard for the group limit sets a cumulative limit for all users in that queue. 
+If there is no specific group mentioned the wildcard group limit would thus be the same as the queue limit. 
+For that reason we do not allow specifying only one group limit that is using the wildcard. 
+There must be at least one limit with a group name defined.
+
+The combination of one or more named group limits with a wildcard group limit is allowed. 
+There is no requirement to have a wildcard entry.
+
+The hierarchy is considered bottom up, starting from the leaf up to the root.
+The first group that is specified in the queue hierarchy for limiting usage that matches a group in the _UserGroup_ object is selected for that application. 
+That definition has an impact on the wildcard interpretation and the wildcard has an impact on the group selection. 
+The wildcard will indirectly affect group quotas set for a specific group higher up in the hierarchy. 
+This point needs clear documentation as part of the group selection algorithm and its side effects.
+
+### Example configuration
+
+An example of a set of limits. A mixture of a user and multiple groups each with a limit.
+Followed by the catch-all for both users and groups with separate resource limits.
+
+In the example below all users, except the user “sue” have the same limit set. Members of the group “development” and the group “test” both have the same limit set. All other users who are not a member of the group “development” or “test” are counted towards the cumulative group limit.
+```yaml
+limits:
+- limit: "specific user"
+  users:
+  - "sue"
+  maxresources: {memory: 25G, vcore: 5}
+- limit: "specific groups"
+  groups:
+  - "development"
+  - "test"
+  maxresources: {memory: 100G, vcore: 10}
+- limit: "user catch all"
+  users:
+  - "*"
+  maxresources: {memory: 10G, vcore: 1}
+- limit: "group catch all"
+  groups:
+  - "*"
+  maxresources: {memory: 50G, vcore: 10}
+```
+
+### Syntax checking
+
+As part of the already existing queue configuration processing the syntax for the configuration is checked. 
+This check is triggered also for a reload of the file, via the config map, or the REST interface.
```

Comment:
Hi @wilfred-s , if we only can change via config map from 1.2.0 version, if we need to change the description here.

Reply:
This document was written before we made that change. This is purely a copy of the content of the google doc without changes to the content. See the original google doc mentioned in the umbrella jira

Reply:
Sure @wilfred-s , it makes sense to me, thank you for clarify!